src/compiler/checker.ts

```diff
@@ -167,10 +167,10 @@ namespace ts {
 
         return checker;
 
-        function getEmitResolver(sourceFile?: SourceFile) {
+        function getEmitResolver(sourceFile: SourceFile, cancellationToken: CancellationToken) {

```

Comment:
I believe we already have an implementation for cancellation. All i am saying we do not need two ways. i personally like the current model as it does not sprinkle the cancellation token through different interfaces. the argument about inconsistent state of the checker is more of an implementation detail that we should look into solving. i am not convinced that adding an additional way of setting cancellation tokens is the right solution.